frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import { ALL_NAMESPACES_KEY } from '@console/shared';
+import { ProjectRequestModel } from '@console/internal/models';
+import { RootState } from '@console/internal/redux';
+import { connect } from 'react-redux';
+
+import { k8sCreate } from '@console/internal/module/k8s';
+import {
+  CloudShellSetupFormData,
+  CREATE_NAMESPACE_KEY,
+  cloudShellSetupValidation,
+} from './cloud-shell-setup-utils';
+import CloudSehellSetupForm from './CloudShellSetupForm';
+import { WorkspaceModel } from '../../../models';
+import { newCloudShellWorkSpace, createCloudShellResourceName } from '../cloud-shell-utils';
+
+interface StateProps {
+  activeNamespace: string;
+  username: string;
+}
+
+type Props = StateProps & {
+  onSubmit?: () => void;
+  onCancel?: () => void;
+};
+
+const CloudShellSetup: React.FunctionComponent<Props> = ({
+  activeNamespace,
+  onSubmit,
+  onCancel,
+  username,
+}) => {
+  const initialValues: CloudShellSetupFormData = {
+    namespace: activeNamespace === ALL_NAMESPACES_KEY ? undefined : activeNamespace,
+  };
+  const handleSubmit = async (values: CloudShellSetupFormData, actions) => {
+    actions.setStatus({ isSubmitting: true });
+    const createNamespace = values.namespace === CREATE_NAMESPACE_KEY;
+    const namespace = createNamespace ? values.newNamespace : values.namespace;
+
+    try {
+      if (createNamespace) {
+        await k8sCreate(ProjectRequestModel, {
+          metadata: {
+            name: namespace,
+          },
+        });
+      }
+      await k8sCreate(
+        WorkspaceModel,
+        newCloudShellWorkSpace(createCloudShellResourceName(), namespace, username),
+      );
+      onSubmit && onSubmit();
+    } catch (err) {
+      actions.setSubmitting(false);
+      actions.setStatus({ submitError: err.message });
```

Comment:
Probably a good idea to also set `isSubmitting: false` in status. And then you won't need `actions.setSubmitting(false)`